src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserReleaseRoleRepositoryTests.cs

```diff
@@ -69,7 +68,96 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
+                    .ToListAsync();
+                Assert.Single(userReleaseRoles);
+
+                Assert.NotEqual(Guid.Empty, userReleaseRoles[0].Id);
+                Assert.Equal(userId, userReleaseRoles[0].UserId);
+                Assert.Equal(releaseId, userReleaseRoles[0].ReleaseId);
+                Assert.Equal(Contributor, userReleaseRoles[0].Role);
+                Assert.Equal(createdById, userReleaseRoles[0].CreatedById);
+                Assert.InRange(DateTime.UtcNow.Subtract(userReleaseRoles[0].Created!.Value).Milliseconds,
+                    0, 1500);
+            }
+        }
+
+        [Fact]
+        public async Task CreateIfNotExists_AlreadyExists()
+        {
+            var userReleaseRole = new UserReleaseRole
+            {
+                UserId = Guid.NewGuid(),
+                ReleaseId = Guid.NewGuid(),
+                Role = Lead,
+                CreatedById = Guid.NewGuid(),
+                Created = new DateTime(2021, 12, 25),
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddAsync(userReleaseRole);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var service = SetupUserReleaseRoleRepository(contentDbContext);
+
+                var result = await service.CreateIfNotExists(userReleaseRole.UserId,
+                    userReleaseRole.ReleaseId, userReleaseRole.Role, userReleaseRole.CreatedById.Value);
+
+                Assert.NotEqual(Guid.Empty, result.Id);
+                Assert.Equal(userReleaseRole.UserId, result.UserId);
+                Assert.Equal(userReleaseRole.ReleaseId, result.ReleaseId);
+                Assert.Equal(userReleaseRole.Role, result.Role);
+                Assert.Equal(userReleaseRole.CreatedById, result.CreatedById);
+                Assert.Equal(userReleaseRole.Created, result.Created);
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var userReleaseRoles = await contentDbContext.UserReleaseRoles
+                    .AsQueryable()
+                    .ToListAsync();
+                Assert.Single(userReleaseRoles);
+
+                Assert.NotEqual(Guid.Empty, userReleaseRoles[0].Id);
+                Assert.Equal(userReleaseRole.UserId, userReleaseRoles[0].UserId);
+                Assert.Equal(userReleaseRole.ReleaseId, userReleaseRoles[0].ReleaseId);
+                Assert.Equal(userReleaseRole.Role, userReleaseRoles[0].Role);
+                Assert.Equal(userReleaseRole.CreatedById, userReleaseRoles[0].CreatedById);
+                Assert.Equal(userReleaseRole.Created, userReleaseRoles[0].Created);
```

Comment:
It would be simpler to just `var userReleaseRole = Assert.Single(contentDbContext.UserReleaseRoles);` and then use `userReleaseRole` rather than `userReleaseRoles[0]`.